crates/sui-types/src/crypto.rs

```diff
@@ -1368,6 +1368,9 @@ where
         // Remove name tag before deserialization using BCS
         let name = serde_name::trace_name::<Self>().expect("Self should be a struct or an enum");
         let name_byte_len = format!("{}::", name).bytes().len();
```

Comment:
Also a short optimization for the future is that to compute `name_len_byte` we could just do `name.len() + 2`, no need to rewrite it